v1/maas.yaml

```diff
@@ -0,0 +1,96 @@
+description: A local MAAS server with LXD VM host built in.
+version: latest
+
+instances:
+  name: maas
+  limits:
+    min-cpu: 4
+    min-mem: 8G
+    min-disk: 30G
+
+  ssh_pwauth: yes
+  chpasswd:
+    list: |
+      multipass:multipass
+    expire: False
+  write_files:
+  - content: |
+      config:
+        core.https_address: '[::]:8443'
+        core.trust_password: password
+      networks:
+      - config:
+          ipv4.address: 10.10.10.1/24
+          ipv6.address: none
+        description: ""
+        name: lxdbr0
+        type: ""
+        project: default
+      storage_pools:
+      - config:
+          size: 16GB
+        description: ""
+        name: default
+        driver: zfs
+      profiles:
+      - config: {}
+        description: ""
+        devices:
+          eth0:
+            name: eth0
+            network: lxdbr0
+            type: nic
+          root:
+            path: /
+            pool: default
+            type: disk
+        name: default
+      projects: []
+      cluster: null
+    path: /tmp/lxd.cfg
+  
+  packages:
+    jq
+  snap:
+    commands:
+      - snap install maas
+      - snap install lxd
+      - snap install maas-test-db
+  runcmd:
+  - echo "Enabling IPv4 forwarding and NAT for VMs bridge"
+  - export IP_ADDRESS=$(ip -4 addr show ens4 | grep -oP '(?<=inet\s)\d+(\.\d+){3}')
+  - sysctl -w net.ipv4.ip_forward=1
+  - iptables -t nat -A POSTROUTING -o ens4 -j SNAT --to $IP_ADDRESS
+  - echo iptables-persistent iptables-persistent/autosave_v4 boolean true | sudo debconf-set-selections
+  - echo iptables-persistent iptables-persistent/autosave_v6 boolean true | sudo debconf-set-selections
+  - apt-get install iptables-persistent -y
+  - cat /tmp/lxd.cfg | lxd init --preseed
+  - echo "Sleeping for 20 seconds"
+  - sleep 20
+  - echo "Initing maas"
+  - maas init region+rack --database-uri maas-test-db:/// --maas-url http://127.0.0.1:5240/MAAS
+  - echo "Init done"
+  - maas createadmin --username admin --password admin --email admin
+  - export APIKEY=$(maas apikey --username admin)
+  - maas login admin 'http://localhost:5240/MAAS/' $APIKEY
+  - echo "Sleeping for 15 seconds"
```

Comment:
But it isn't _actually_ sleeping here :)

It's just saying it is, but there's no `sleep 15` below :)

And the bigger problem is that it takes 15 seconds on your machine. But on a slower one it may take 20s, and the steps below will fail.

What I tend to do is something along the lines of `timeout <big_timeout> sh -c "until <condition>; do sleep 1; done"`.

You will need to work out what `<condition>` should be. Maybe it's connecting to a TCP socket, maybe it's grepping through a `curl` output – it will depend on the use case.

Reply:
NB: I mean, after MAAS init, which, after running - still appears to be doing things in the background and MAAS takes awhile to settle.

Reply:
OK, I understand what you're saying and I thought that too, it takes longer on other people's machines (or shorter). I don't know how to test the MAAS working condition yet. I'll have a think about it.

Reply:
funny that I only had a comment :)